core/opendaq/device/include/opendaq/device_impl.h

```diff
@@ -1660,6 +1688,44 @@ void GenericDevice<TInterface, Interfaces...>::deserializeCustomObjectValues(con
     else
         userLock.forceUnlock();
 
+    if (serializedObject.hasKey("connectionStatuses"))
+    {
+        connectionStatusContainer = serializedObject.readObject(
+            "connectionStatuses",
+            context,
+            [this](const StringPtr& typeId,
```

Comment:
1) Can you move this code above the deserialization of "UserLock" filed? It is probably safer if the user lock is the last field to be deserialized.

2) This looks like duplicated code to me. I suspect you are doing it this way, because you want to trigger the core event while deserializing? Would it be possible to use deserialize context parameter to pass the core event callback to ConnectionStatusContainerImpl::Deserialize method and have all the code there?

Reply:
1. Done. Could it be also unsafe that default folders and components are being deserialized after UserLock ?

2. Yes, the code in this deserialization method of device duplicates the one in ConnectionStatusContainer deserialization static method. Similarly, same approach is used for the ComponentStatusContainer and component Tags to pass the coreEvent callback into the constructors.
I think, it would be good to eliminate these duplications as well, though not sure if the deserialize context supports passing any type of parameter. Does it ?

Reply:
2. The ComponentDeserializationContext has been extended to support specifying a triggerCoreEvent procedure for component. It is now utilized to eliminate code duplication in the deserialization of status containers and tags.